### DISCUSSION TITLE
Update PrismaClient while developing NextJS

### DISCUSSION THREAD

**Discussion by conduongtong** [prisma/prisma#22405] [Q&A] [answered]
### Question

So I followed this article to implement Prisma with NextJS: https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/nextjs-prisma-client-dev-practices

```ts
import { PrismaClient } from '@prisma/client'

const prismaClientSingleton = () => {
  return new PrismaClient()
}

declare global {
  var prisma: undefined | ReturnType<typeof prismaClientSingleton>
}

const prisma = globalThis.prisma ?? prismaClientSingleton()

export default prisma

if (process.env.NODE_ENV !== 'production') globalThis.prisma = prisma
```

When I clicked a button, it will run ```exec('npx prisma db push')```.
Prisma will update but 'const prisma' still use old one, how can I update it without turn off NextJS while developing?

**Comment by ludralph** [comment]
Hi @conduongtong 👋 

Thank you for raising this question.

Do you want to run Prisma migrate commands like `db push` programmatically without involving CLI?

> Prisma will update, but 'const prisma' still use old one, how can I update it without turn off NextJS while developing

Can you please elaborate on this statement?

**Reply by conduongtong** [reply, marked as answer]
I found a solution by using nodemon --exec 'npx prisma db push && next dev'. When clicking a button, it will update the restart.json file, and nodemon will detect restart.json changes to restart next dev.
